### PR TITLE
Creating Content Author for CreatePageAsAuthorUserIT

### PR DESCRIPTION
**Pull request by absarasw**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by absarasw** [comment]
@dulvac, @volteanu please review the PR

File: smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/CreatePageAsAuthorUserIT.java [hunk 1]
```diff
@@ -16,48 +16,91 @@
...
+    private CQClient testUserClient;
 
-    // Create a random page so the test site is initialized properly.
-    private final Page temporaryPage = new Page(cqBaseClassRule.authorRule);
 
-    public TemporaryUser userRule = new TemporaryUser(() -> cqBaseClassRule.authorRule.getAdminClient(), CONTENT_AUTHORS_GROUP);
+    @Before
+    public void setupSecurityBase() throws ClientException, InterruptedException {
+        adminAuthor = cqBaseClassRule.authorRule.getAdminClient(CQClient.class);
+        authorAuthor = cqBaseClassRule.authorRule.getClient(CQClient.class, "author", "author");
```
Comment (dulvac) [inline smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/CreatePageAsAuthorUserIT.java]: @absarasw , we cannot make an assumption that this user exists. A new one needs to be created, hence the `TemporaryUser` rule.

File: smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/CreatePageAsAuthorUserIT.java [hunk 2]
```diff
@@ -16,48 +16,91 @@
...
+
+    private CQSecurityClient sClient;
+    private CQPermissions permissionsObj;
+
+    private User testUser;
+    private Group testGroup;
+    private CQClient testUserClient;
 
-    // Create a random page so the test site is initialized properly.
-    private final Page temporaryPage = new Page(cqBaseClassRule.authorRule);
```
Comment (dulvac) [inline smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/CreatePageAsAuthorUserIT.java:55]: Why do we need to remove this? We should use Rules and not before and after methods, for modularity.

File: smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/CreatePageAsAuthorUserIT.java [hunk 3]
```diff
@@ -68,16 +111,16 @@
     @Test
     public void testCreatePageAsAuthor() throws InterruptedException, ClientException {
         String pageName = "testpage_" +  UUID.randomUUID();
-        String pagePath = temporaryPage.getParentPath() + "/" + pageName;
+        String pagePath = "/content/" + pageName;
```
Comment (dulvac) [inline smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/CreatePageAsAuthorUserIT.java]: This is really not the way to go, creating pages directly under `/content` or hardcode it. The `Page` rule has a homogenous way of creating content under a test site it always provisions if missing. These tests should be able to run on any instance of AEM, with no assumptions about existing users and they need to be safe to execute against a production environment.

**Review by dulvac — CHANGES_REQUESTED** [review]
Some comments inline. Also, may I ask what issue prompted this PR?

**Comment by absarasw** [comment]
>   may I ask what issue prompted this PR?

As mentioned over the jira issues https://jira.corp.adobe.com/browse/SKYOPS-13827 and https://jira.corp.adobe.com/browse/SITES-1013, some of the customers have removed the write access for **content-authors** under '/content' path. So we have to create a separate user and group and assign the correct ACLs to write under '/content' path.

File: smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/CreatePageAsAuthorUserIT.java (hunk 1 above)
Comment (absarasw) [reply smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/CreatePageAsAuthorUserIT.java]: This shouldn't be there, I'll remove this.

File: smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/CreatePageAsAuthorUserIT.java (hunk 2 above)
Comment (absarasw) [reply smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/CreatePageAsAuthorUserIT.java:55]: I was wondering how can we create a new user and group and assign correct ACLs using rules. Let me know if there is a way.

File: smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/CreatePageAsAuthorUserIT.java (hunk 3 above)
Comment (absarasw) [reply smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/CreatePageAsAuthorUserIT.java]: I was thinking the same, but the test site is not getting initialised without the `cqRuleChain` rule which is dependant on `TemporaryUser` rule.

**Comment by absarasw** [comment]
@dulvac , @volteanu I have updated the test case with TemporaryContentAuthorGroup Rule added in https://github.com/adobe/aem-testing-clients/pull/58. Please review now.

**Comment by absarasw** [comment]
@dulvac can you review the PR?

**Review by dulvac — APPROVED** [review]

**Comment by dulvac** [comment]
@absarasw, thanks for the contribution! Looks nice and clean now.

**Pull request merged**